src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -2059,6 +2063,94 @@ def aks_upgrade(cmd,    # pylint: disable=unused-argument, too-many-return-state
+    if mc.aad_profile is not None and mc.aad_profile.managed:
+        request_payload.cluster_token = _get_dataplane_aad_token(cmd.cli_ctx, "6dae42f8-4368-4678-94ff-3960e28e3630")
+
+    commandResultFuture = client.run_command(resource_group_name, name, request_payload, long_running_operation_timeout=5, retry_total=0)
+    commandResult = commandResultFuture.result(300)
+    _print_command_result(commandResult)
+
+
+def aks_command_result(cmd, client, resource_group_name, name, command_id=""):
+    if not command_id:
+        raise CLIError('CommandID cannot be empty.')
+
+    commandResult = client.get_command_result(resource_group_name, name, command_id)
+    _print_command_result(commandResult)
+
+
+def _print_command_result(commandResult):
+    # succeed, print exitcode, and logs
+    if commandResult.provisioning_state == "Succeeded":
+        print(f"{colorama.Fore.GREEN}command started at {commandResult.started_at}, finished at {commandResult.finished_at}, with exitcode={commandResult.exit_code}{colorama.Style.RESET_ALL}")
+        print(commandResult.logs)
+        return
+    # failed, print reason in error
+    if commandResult.provisioning_state == "Failed":
+        print(f"{colorama.Fore.RED}command failed with reason: {commandResult.reason}{colorama.Style.RESET_ALL}")
+        return
+    # *-ing state
+    print(f"{colorama.Fore.BLUE}command is in : {commandResult.provisioning_state} state{colorama.Style.RESET_ALL}")
+
+
+def _get_command_context(command_files):
+    if not command_files:
+        return ""
+
+    filesToAttach = {}
+    # . means to attach current folder, cannot combine more files. (at least for now)
+    if len(command_files) == 1 and command_files[0] == ".":
+        # current folder
+        cwd = os.getcwd()
+        for filefolder, _, files in os.walk(cwd):
+            for file in files:
+                # retain folder structure
+                rel = os.path.relpath(filefolder, cwd)
+                filesToAttach[os.path.join(filefolder, file)] = os.path.join(rel, file)
+    else:
+        for file in command_files:
+            if file == ".":
+                raise CLIError(". is used to attach current folder, not expecting other attachements.")
+            if os.path.isfile(file):
+                # for individual attached file, flatten them to same folder
+                filesToAttach[file] = os.path.basename(file)
+
+    zipStream = io.BytesIO()
+    zipFile = zipfile.ZipFile(zipStream, "w")
+    for _, (osfile, zipEntry) in enumerate(filesToAttach.items()):
+        zipFile.write(osfile, zipEntry)
+    # zipFile.printdir() // use this to debug
+    zipFile.close()
+
+    return str(base64.encodebytes(zipStream.getbuffer()), "utf-8")
+
+
+def _get_dataplane_aad_token(cli_ctx, serverAppId):
+    # this function is mostly copied from keyvault cli
+    import adal
+    try:
+        return Profile(cli_ctx=cli_ctx).get_raw_token(serverAppId)[0][2].get('accessToken')
```

Comment:
[0][2]? and we make sure it won't out of index?

Reply:
1. this is copied from keyvault CLI, I assume this is well tested.
2. it does have a try block covers it.